packages/nx/src/plugins/js/package-json/create-package-json.ts

```diff
@@ -177,6 +177,10 @@ export function createPackageJson(
     packageJson.peerDependenciesMeta
   );
 
+  if (rootPackageJson.packageManager) {
```

Comment:
Having it use the root package.json by default makes a lot of sense to me, but I could imagine edge cases where the generated package.json, and the environment in which it is used, do not match the environment of the workspace 1:1.

To allow for users to handle such a case I think it would be make sense to preferentially respect the `packageManager` field on the project package.json (if the file and field exist) and use that when generating.

What do you think @FrozenPandaz?

Reply:
Respecting local package.json seems reasonable but also leaves room for user to specify incompatible package manager, which would then mess up the lock file generation.

The root package.json package manager is the one we used for generation and testing and therefore the only one for which we can guarantee things to work.